### PR TITLE
NMOS Controller specification

### PR DESCRIPTION
**Pull request by garethsb**
Squashed version of https://github.com/AMWA-TV/is-05/pull/148 for publication review.

### CODE REVIEW

**Comment by garethsb** [comment]
There's a the discrepancy here between the final sentence of the **Connection Management** section:

> When connecting a Receiver to an NMOS Sender, the Controller SHOULD communicate the Sender UUID to the receiver using the `sender_id` parameter of the `/staged` endpoint.

and the section **Identifying Active Connections** which says this:

>  The Controller MUST set and unset (using `null`) the `sender_id` or `receiver_id` parameters when modifying the `transport_params` or `transport_file`.

and the existing wording on the **Interoperability with IS-04** page of the spec:

> In order to populate the subscription attribute of IS-04 Senders and Receivers, the Connection API includes keys for `sender_id` and `receiver_id` in its staged parameters. These SHOULD be used to signal that a Sender or Receiver is being connected to another NMOS compatible Sender or Receiver. It is the client's responsibility to set and unset (using `null`) the `sender_id` or `receiver_id` parameters when modifying the `transport_params` or `transport_file`.

I think that the intention is to strengthen the requirement on Controllers so that they MUST correctly set `sender_id` and `receiver_id` when they know they're configuring a Receiver or Sender to connect to an NMOS Sender or Receiver as opposed to just configuring `transport_params` or `transport_file`, e.g. to point to a non-NMOS Sender/Receiver, in which case `sender_id` or `receiver_id` MUST be set to `null`.

To clarify this, I propose to move the paragraph from **Identifying Active Connections** into the **Connection Management** section (it's really nothing to do with interop with IS-04), replacing the single sentence there now.

We could also modify the existing wording on the **Interoperability with IS-04** page of the spec? Something like:

```diff
< In order to populate the subscription attribute of IS-04 Senders and Receivers, the Connection API includes
< keys for `sender_id` and `receiver_id` in its staged parameters. These SHOULD be used to signal that a
< Sender or Receiver is being connected to another NMOS compatible Sender or Receiver. It is the client's
< responsibility to set and unset (using `null`) the `sender_id` or `receiver_id` parameters when modifying
< the `transport_params` or `transport_file`.
----
> In order to populate the subscription attribute of IS-04 Senders and Receivers, the Connection API includes
> keys for `sender_id` and `receiver_id` in its `/staged` and `/active` endpoints. It is the client's
> responsibility to set and unset (using `null`) the `sender_id` or `receiver_id` parameters when modifying
> the `transport_params` or `transport_file` to signal if a Receiver or Sender is being connected to another
> NMOS compatible Sender or Receiver.
```

**Review by pedro-alves-ferreira — APPROVED** [review]

File: docs/Overview.md [hunk 1]
```diff
@@ -21,7 +21,8 @@ The Specification includes:
 
 IS-05 is intended to be used in conjunction with an [IS-04 NMOS Discovery & Registration](https://specs.amwa.tv/is-04) deployment; however it has been written in such a way to provide useful functionality even in the absence of such a system.
 
-The terms 'Node', 'Device', 'Sender' and 'Receiver' are used extensively in this documenation set. The [NMOS Technical Overview](https://specs.amwa.tv/nmos/main/docs/2.0._Technical_Overview.html) provides an outline of these terms, and IS-04 provides corresponding schema definitions.
+The terms 'Node', 'Device', 'Sender' and 'Receiver' are used extensively in this documentation set.
+The [NMOS Technical Overview](https://specs.amwa.tv/nmos/main/docs/2.0._Technical_Overview.html) and the [NMOS Glossary][Glossary] define these and other common terms that have specific meanings in NMOS.
```
Comment (peterbrightwell) [inline docs/Overview.md]:
```suggestion
The [NMOS Technical Overview](https://specs.amwa.tv/nmos/main/docs/Technical_Overview.html) and the [NMOS Glossary][Glossary] define these and other common terms that have specific meanings in NMOS. 

```

Comment (garethsb) [reply docs/Overview.md]: Thanks, @peterbrightwell. That link presumably needs fixing in other specs and other branches, and will be broken in all previously tagged versions. 😒 With a quick search, I think that means all previous published versions of IS-05 and IS-08...

**Review by peterbrightwell — APPROVED** [review]
LGTM (didn't check links)

**Pull request merged**